modules/stochastic_tools/include/state_sim/model/actions/StateSimAction.h

```diff
@@ -0,0 +1,38 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef STATESIMACTION_H
+#define STATESIMACTION_H
+
+#include "StateSimBase.h"
+#include <string>
+
+class StateSimAction : public StateSimBase
+{
+protected:
+  ACTION_TYPE_ENUM _act_type;
+
+private:
+  static unsigned int _counter;
```

Comment:
The problem with statics is you get one copy per process. It basically is a global variable. That means any kind of write access "should" be guarded or locked, or guaranteed not to occur in a threaded region. It might be that way today but it doesn't take long before somebody makes a mistake and you end up with a really nasty race condition bug. There are other ways of generating unique IDs, or we could look into more bulletproof ways of using this in a threaded region. Maybe we should explore the "atomics" in C++11.